### ISSUE TITLE
Allow join conditions to use an object.

### ISSUE THREAD

**Issue by whitelynx** [tgriesser/knex#743]
Currently, join conditions may be specified in 3 ways:
```javascript
knex.from('users').innerJoin('accounts', 'users.id', 'accounts.user_id')

knex.table('users').innerJoin('accounts', 'users.id', '=', 'accounts.user_id')

knex('users').innerJoin('accounts', function() {
  this.on('accounts.id', '=', 'users.account_id').orOn('accounts.owner_id', '=', 'users.id')
})
```

However, those familiar with `.where()` would expect a 4th possibility:
```javascript
knex('users').innerJoin('accounts', {
  'accounts.id': 'users.account_id',
  'accounts.owner_id': 'users.id'
})
```

Could this syntax be added?

**Comment by bendrucker** [comment]
For sure. We'll definitely take a PR.

**Comment by tgriesser** [comment]
Wow that's such an obvious/better syntax. I'll definitely throw it on the list.

**Comment by tgriesser** [comment]
Available in 0.8
